### PR TITLE
Add CRA example with Middleware.

### PR DESCRIPTION
**Pull request by leerob**

### CODE REVIEW

File: edge-functions/create-react-app/src/vitals.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    'effectiveType' in navigator['connection']
+    ? navigator['connection']['effectiveType']
+    : '';
+}
+
+export function sendToVercelAnalytics(metric) {
+  const analyticsId = process.env.REACT_APP_VERCEL_ANALYTICS_ID;
+  if (!analyticsId) {
+    return;
+  }
```
Comment (lfades) [inline edge-functions/create-react-app/src/vitals.js:15]: Should we add instructions on how to set up this?

File: edge-functions/create-react-app/src/vitals.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    dsn: analyticsId,
+    id: metric.id,
+    page: window.location.pathname,
+    href: window.location.href,
+    event_name: metric.name,
+    value: metric.value.toString(),
+    speed: getConnectionSpeed(),
+  };
+
+  console.log({ body });
```
Comment (lfades) [inline edge-functions/create-react-app/src/vitals.js:27]:
```suggestion
```

File: edge-functions/create-react-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import './App.css';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <p>
+          Edit <code>src/App.js</code> and save to reload.
+        </p>
```
Comment (lfades) [inline edge-functions/create-react-app/src/App.js:11]: Wdyt about adding another paragraph saying "Open devtools and you should be able to see `x-cra-example` in the list, added by middleware.js".

**Review by lfades — CHANGES_REQUESTED** [review]
Added some comments!

Can we add TS support to the example (with `strict: false` if it asks for too much typing)? as I find it unlikely we'll have a duplicated version of this example with TS support in the future.

File: edge-functions/create-react-app/src/vitals.js (hunk 1 above)
Comment (leerob) [reply edge-functions/create-react-app/src/vitals.js:15]: https://vercel.com/docs/concepts/analytics/api

**Pull request closed without merging**